pkg/common/contexts.go

```diff
@@ -0,0 +1,30 @@
+package common
+
+import (
+	"context"
+	"sync"
+)
+
+func MergeContexts(ctxs ...context.Context) (context.Context, context.CancelFunc) {
+	ctx, cancel := context.WithCancel(context.Background())
+	var wg sync.WaitGroup
+
+	for _, c := range ctxs {
```

Comment:
also are there no built-ins for this kind of functionality? feels like overkill to add a separate utility for it

Reply:
nvm not sure that one does what we want here

Reply:
I don't think there is a built-in for this. There are 3rd party packages, but many of them seem to manage state with a struct and locks. To me it seems like overkill to manage state for this.